Лекция 9

entgorbunov opened the pull request:
Домашнее задание:
Повторить у себя локально всё то что было сделано в лекции:
postgres в docker
Spring Data JPA
liquibase
AuditableEntity
инкремент количества открытий короткой ссылки
эндпоинт поиска по фильтру
Заменить ошибку NotFoundException при получении короткой ссылки на NotFoundShortLinkException

entgorbunov committed:
Лекция 9

Домашнее задание:
Повторить у себя локально всё то что было сделано в лекции:
postgres в docker
Spring Data JPA
liquibase
AuditableEntity
инкремент количества открытий короткой ссылки
эндпоинт поиска по фильтру
Заменить ошибку NotFoundException при получении короткой ссылки на NotFoundShortLinkException

panyukovnn commented on src/main/java/com/panyukovnn/linkshortener/controller/LinkInfoController.java:
```diff
@@ -41,11 +45,37 @@ public CommonResponse<LinkInfoResponse> createShortLink(@RequestBody @Valid Comm
+            .build();
+    }
+
+    @PutMapping()
```
Скобки после аннотации не нужны

panyukovnn replied:
И это у нас Patch метод

entgorbunov replied:
done

panyukovnn commented on src/main/java/com/panyukovnn/linkshortener/controller/LinkInfoController.java:
```diff
@@ -41,11 +45,37 @@ public CommonResponse<LinkInfoResponse> createShortLink(@RequestBody @Valid Comm
+    }
+
     @GetMapping
     public CommonResponse<List<LinkInfoResponse>> getAllLinks() {
```
Вот этот эндпоинт не нужен, т.к. при запросе по фильтру без параметров фильтрации - будут возвращаться все LinkInfo

entgorbunov replied:
done

panyukovnn commented on src/main/java/com/panyukovnn/linkshortener/repository/LinkInfoRepository.java:
```diff
@@ -1,21 +1,51 @@
+        FROM LinkInfo
+                WHERE shortLink = :shortLink
+                        AND active = true
+                                AND (endTime is null or endTime > :now)
```
Не стоит делать таких отступов в запросах, SELECT / FROM / WHERE и т.д. все на одном уровне, если это конечно не под-запросы

entgorbunov replied:
done

panyukovnn commented on src/main/java/com/panyukovnn/linkshortener/service/impl/LinkInfoServiceImpl.java:
```diff
@@ -59,19 +77,16 @@ public LinkInfoResponse createLinkInfo(CreateShortLinkRequest request) {
-            linkInfo.setLink(request.getLink());
-        }
+
+        linkInfo.setDescription(request.getDescription());
```
Оставь условия на присваивание, только при != null, т.к. это метод частичного обновления полей

entgorbunov replied:
done

entgorbunov committed:
- Убраны лишние отступы в SQL запросах
- Исправлен тип HTTP метода с PUT на PATCH для частичного обновления
- Удалены избыточные условия присваивания в PATCH методе
- Удален эндпоинт getAllLinks
- Убраны пустые скобки в маппинг аннотациях

panyukovnn commented on src/main/java/com/panyukovnn/linkshortener/repository/LinkInfoRepository.java:
```diff
@@ -1,21 +1,51 @@
+
+    @Query("""
+        UPDATE LinkInfo
+                set openingCount = openingCount + 1
```
Отступы выровняй во всех остальных запросах тоже

И старайся придерживаться единого стиля, в данном случае set и where надо прописать заглавными буквами

entgorbunov replied:
done

panyukovnn commented on src/main/java/com/panyukovnn/linkshortener/service/impl/LinkInfoServiceImpl.java:
```diff
@@ -70,8 +89,12 @@ public LinkInfoResponse updateLinkInfo(UpdateShortLinkRequest request) {
             linkInfo.setLink(request.getLink());
         }
-        linkInfo.setEndTime(request.getEndTime());
+        if (request.getEndTime() != null) {
```
Уже говорил что endTime может быть null, поэтому тут проверка на != null не нужна

entgorbunov replied:
done

entgorbunov committed:
правки
